pkg/kubelet/cm/event_dispatcher.go

```diff
@@ -307,37 +307,49 @@ func ResourceConfigFromReply(reply *lifecycle.EventReply, resources *ResourceCon
 
 // Updates the supplied container config in-place based on the isolation
 // controls in the event reply.
-func UpdateContainerConfigWithReply(reply *lifecycle.EventReply, config *runtime.ContainerConfig) {
+func UpdateContainerConfigWithReply(reply *lifecycle.EventReply, config *runtime.ContainerConfig) *runtime.ContainerConfig {
+	if reply == nil {
+		glog.Info("eventDispatcherNoop has been detected - skipping container configuration update")
+		return config
+	}
+	if reply.Error != "" {
+		glog.Errorf("isolator returned error: %s", reply.Error)
+		return config
+	}
+
+	updatedConfig := &runtime.ContainerConfig{}
+	*updatedConfig = *config
```

Comment:
This is not a safe copy. `runtime.ContainerConfig` is a struct with reference members; copying the references into a new struct and then modifying the memory they point to effectively modifies the initial value in place. In `ResourceConfigFromReply` this was coincidentally safe to do because the struct contains only pointers to primitive types, and we happen to only assign new pointers in update (and not update the memory those pointers point to).

If we want consistency, let's just make both of these functions update in-place and return only an error.